### ISSUE TITLE
AutoFit: Update map focus when main item changes location

### ISSUE THREAD

**Issue by jaswalters** [nathan-gs/ha-map-card#86]
Add an AutoFit option like the standard map has. Possible?

**Comment by nathan-gs** [comment]
In the latest [v1.6.0](https://github.com/nathan-gs/ha-map-card/releases/tag/v1.6.0) this has been added by @baeksj.

**Comment by jaswalters** [comment]
I was expecting it to move the map like the HA card does when a location changes but did not on my end. Is my expectation incorrect?

**Comment by nathan-gs** [comment]
@jaswalters fixed in [v1.12.0](https://github.com/nathan-gs/ha-map-card/releases/tag/v1.12.0) (with the follow_focus option).

**Comment by jaswalters** [comment]
Thanks so much! Will give it a try when time permits and report back.

**Comment by bikeymouse** [comment]
This seems not to work for my. Wonder If I'm the only one?
For central tracking of this issue, see also [See HomeAssistant forum](https://community.home-assistant.io/t/map-card-a-slightly-improved-map-card/693088/68?u=bikey) for my experience.

**Comment by nathan-gs** [comment]
See #125

**Comment by nathan-gs** [comment]
@jaswalters @bikeymouse there was an error in the README, it's fixed in [v1.13.1](https://github.com/nathan-gs/ha-map-card/releases/tag/v1.13.1). In short the config is named `focus_follow`.

**Comment by bikeymouse** [comment]
Yes, the focus is now updated properly. Thanks for the quick update!

Found another strange behaviour though.
It seems that the focus_follow options messes with the zoom functionality. 
If I change the zoom option to anything higher then 13 it is zooming in to the maximun level (even ignoring the max_zoom)
I'd like to have it to zoom to 15, but it zooms to 20.

See how the zoom jumps between 14 and 15:
With zoom: 13:
![Image](https://github.com/user-attachments/assets/ea710293-c7fe-4145-a471-0904f1bdce22)
With zoom 14:

![Image](https://github.com/user-attachments/assets/dd21445b-7c43-46b1-8e17-89e7493c38aa)
While without the focus_follow option the zoom:15 is as it should:

![Image](https://github.com/user-attachments/assets/00b162d6-2c1e-4528-85d7-5f95ff20cb6c)

**Comment by bikeymouse** [comment]
Btw: it tried this both with `focus_follow: refocus` and `focus_follow: contains`
Don't actually understand the difference, that could perhaps also be added in the documenation ;-)

**Comment by nathan-gs** [comment]
@bikeymouse 
- refocus zooms to fit the entities
- contains does nothing if it fits, follows the refocus behavior if it doesn't

So correct zoom might in both cases be broken.

**Comment by bikeymouse** [comment]
> [@bikeymouse](https://github.com/bikeymouse)
> 
>     * refocus zooms to fit the entities
> 
>     * contains does nothing if it fits, follows the refocus behavior if it doesn't
> 
> 
> So correct zoom might in both cases be broken.

Do you mean that this works as intended from your perspective, or that this is a bug indeed?

Another thing related to the zoom I noticed (when focus_follow: refocus is used), is that when you manually zoom the view in or out, this is quite quickly reset to the default zoom when the window is updated (on a " tick" from Home Assistant), even when the tracked item is stationary.

**Comment by nathan-gs** [comment]
@bikeymouse with `refocus` it's by design it resets the zoom, if that's unwanted use the contains option.

**Comment by bikeymouse** [comment]
> [@bikeymouse](https://github.com/bikeymouse) with `refocus` it's by design it resets the zoom, if that's unwanted use the contains option.

Yeah I would hope so, but also when using the contains option it resets the zoom on the first "tick" of HomeAssistant when manually changed while looking at the dashboard.

And is this behaviour explainable?
> It seems that the focus_follow options messes with the zoom functionality.
If I change the zoom option to anything higher then 13 it is zooming in to the maximun level (even ignoring the max_zoom)
I'd like to have it to zoom to 15, but it zooms to 20.

**Comment by bikeymouse** [comment]
Some other suggestion: the current behaviour of the focus_follow option is that it only "follows" entity when it moves out of the viewport.
What I would prefer is that it would keep the entity in the center of the viewport all the time, just like the standard map-card does with auto_fit
Would that be possible?

**Comment by nathan-gs** [comment]
Shipped as `focus_follow`:

- `refocus` — fit bounds on every update (always keep entities in view; this is the stock-map `auto_fit` equivalent)
- `contains` — only move when something leaves the viewport
- `focus_follow_pause` (#221) — pause follow after the user pans/zooms

`focus_on_fit: false` on an entity excludes it from the fit. The reset-focus button recenters on demand.

Closing as completed. If you still want “keep one entity glued to the exact center without changing zoom”, that is #140 — please continue there.
